### PR TITLE
add further info for device id encoding

### LINKED ISSUE #1231: IoT hub client init
Add documentation for the DeviceID format

### PR DESCRIPTION
**Pull request by danewalton**
Closes #1231

### CODE REVIEW

**Review by ericwol-msft — APPROVED** [review]

File: sdk/inc/azure/iot/az_iot_hub_client.h
Comment (ahsonkhan) [inline sdk/inc/azure/iot/az_iot_hub_client.h:78]: What is the **data format** of the device id? Is it base64 encoded? Is it a Guid? Is it a number?

**Comment by ahsonkhan** [comment]
Please wait until we have a preview5 release before merging.

File: sdk/inc/azure/iot/az_iot_hub_client.h
Comment (danewalton) [reply sdk/inc/azure/iot/az_iot_hub_client.h:78]: I'm not sure if we need to go that far into detail. I think the minute specifics of the documentation are delegated to the service side docs which detail how IoT works in general. The "device id" is a veryyy fundamental and well known construct in IoT. If we are to go to that level of documentation here, tbh it's going to required a bunchhh of work and would be a headache.

Comment (ahsonkhan) [reply sdk/inc/azure/iot/az_iot_hub_client.h:78]: Aside from whether we want to document that or not, for my edification, can you explain what the data format is?

Comment (danewalton) [reply sdk/inc/azure/iot/az_iot_hub_client.h:78]: Sure it's a name given to the device by the user, typically containing mostly letters but could be numbers as well. Standard "C string" encoded as UTF-8 here.

Comment (ahsonkhan) [reply sdk/inc/azure/iot/az_iot_hub_client.h:78]: How long can it be? Can it contain any non-ASCII characters? What about other special characters like `#%/:=?[\]^`{|}`

Comment (ericwol-msft) [reply sdk/inc/azure/iot/az_iot_hub_client.h:78]: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry

A case-sensitive string (up to 128 characters long) of ASCII 7-bit alphanumeric characters plus certain special characters: - . + % _ # * ? ! ( ) , : = @ $ '.

Comment (ahsonkhan) [reply sdk/inc/azure/iot/az_iot_hub_client.h:78]: Is there a link/resource where I can read up on which of these characters need to be percent encoding?

It kind-of looks like the device ID needs to be URL encoded.

**Pull request merged**
